### ISSUE TITLE
Is there a particular reason for disallowing TODO comments?

### ISSUE THREAD

**Issue by tersmitten** [cakephp/cakephp-codesniffer#123]
See [this line](https://github.com/cakephp/cakephp-codesniffer/blob/1.x/CakePHP/ruleset.xml#L20).

In our codebase I would like to allow them. Of course I could just fork this repository, but I just wanted to understand the reason (and maybe propose a change :-)).

**Comment by ravage84** [comment]
@tersmitten you don't need to fork the repo. But you can create a ruleset of your own by extending and/or adjusting an existing rule set.

In your case, it could look like this:

```` xml
<?xml version="1.0"?>
<ruleset name="tersmitten Corp CS">
    <description>Mischa ter Smitten's Company Coding Standard</description>
    <rule ref="../../../cakephp/cakephp-codesniffer/CakePHP">
        <!-- Exclude unwanted sniffs -->
        <exclude name="Generic.Commenting.Todo"/>
    </rule>
    <!-- Add additional, non CakePHP sniffs -->
    <rule ref="Squiz.Commenting.FunctionCommentThrowTag"/>
</ruleset>
````

Put this ruleset.xml into a separate VCS repo and include it through Composer, e.g.:

```` json

    "require-dev": {
        "tersmitten-corp/phpcs": "*"
    },
    "repositories": [
        {
            "type": "vcs",
            "url": "http://example.com/path/to/the/phpcs/ruleset/repo"
        }
    ]
````

I do exactly this with our company's CS. :joy_cat:

**Comment by markstory** [comment]
Instead of a TODO we prefer to just do. :smile: We had problems in CakePHP in the past where people would put TODOs and never do them.

**Comment by tersmitten** [comment]
@markstory, I understand, we have the same problem :-), but still still there're times that we can't fix them now...

**Comment by tersmitten** [comment]
@ravage84, Thanks for the great explanation! I'll try that.

**Comment by tersmitten** [comment]
I've put something together, maybe it's useful for someone else: https://github.com/Oefenweb/cakephp-codesniffer.
